internal/job/integration/checkout_integration_test.go

```diff
@@ -274,6 +275,61 @@ func TestCheckingOutShallowCloneOfLocalGitProject(t *testing.T) {
 	tester.RunAndCheck(t, env...)
 }
 
+func TestCheckingOutLocalGitProjectWithShortCommitHash(t *testing.T) {
+	t.Parallel()
+
+	tester, err := NewBootstrapTester(mainCtx)
+	assert.NilError(t, err)
```

Comment:
(I don't mean to relitigate this here - but I am going to dump some state from my brain because I had to hit the docs for what `assert.NilError` did exactly)

I've been thinking about tests and assert libraries. We now have two (3? 4?) different assert libs in the agent:

* gotest.tools/v3/assert, where the functions call t.FailNow and log maybe too sparsely
* github.com/stretchr/testify/assert, whose functions use t.Errorf and don't terminate the test early, but dump some extra (sometimes redundant) info
* and we also import testify/require in places - the t.Fatalf variation on testify/assert
* bintest and the bootstrap tester 

https://go.dev/play/p/ff8l-kNBibK

Thoughts: 

* The usual argument against assert libraries in general is that it forces us to understand a small extra language with its own nuances in order to work with the assert lib. That we import two different packages in different places called "assert" that have different fail behaviours kinda demonstrates this.
* gotest.tools/v3/assert's functions by default don't include the name of the function that was called that returned the error and what its args were (if not too huge). Putting that info directly in the test log reduces cycle time. But it at least lets us provide that info: `assert.NilError(t, err, "NewBootstrapTester(mainCtx)")`)

Reply:
> Putting that info directly in the test log reduces cycle time.

What happens when the name or arguments to the method changes? I'm pretty sure our code base is littered with examples where that's happened, and now someone reading the test output will be confused. For this reason, I prefer that we don't use this pattern. The file and line of the test failure is always printed, I prefer to start from there.

Reply:
The info being logged is very close to the function call, so we should be better at keeping it updated.